wasm/tests/conftest.py

```diff
@@ -0,0 +1,108 @@
+import subprocess
+import os
+import time
+import socket
+import atexit
+import pytest
+import sys
+
+PORT = 8080
+
+HTTP_SCRIPT = f"""
+import mimetypes
+mimetypes.add_type("application/wasm", ".wasm")
+import http.server
+http.server.test(HandlerClass=http.server.SimpleHTTPRequestHandler, port={PORT})
+"""
+
+demo_dist = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../demo/dist/")
+
+server_proc = None
+
+
+def pytest_sessionstart(session):
+    global server_proc
+    server_proc = subprocess.Popen(
+        ["python3", "-c", HTTP_SCRIPT],
+        cwd=demo_dist,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    wait_for_port(PORT)
+
+
+def pytest_sessionfinish(session):
+    global server_proc
+    server_proc.terminate()
+    server_proc = None
+
+
+atexit.register(lambda: server_proc and server_proc.terminate())
+
+
+# From https://gist.github.com/butla/2d9a4c0f35ea47b7452156c96a4e7b12
+def wait_for_port(port, host="0.0.0.0", timeout=5.0):
+    """Wait until a port starts accepting TCP connections.
+    Args:
+        port (int): Port number.
+        host (str): Host address on which the port should exist.
+        timeout (float): In seconds. How long to wait before raising errors.
+    Raises:
+        TimeoutError: The port isn't accepting connection after time specified in `timeout`.
+    """
+    start_time = time.perf_counter()
+    while True:
+        try:
+            with socket.create_connection((host, port), timeout=timeout):
+                break
+        except OSError as ex:
+            time.sleep(0.01)
+            if time.perf_counter() - start_time >= timeout:
+                raise TimeoutError(
+                    "Waited too long for the port {} on host {} to start accepting "
+                    "connections.".format(port, host)
+                ) from ex
+
+
+from selenium import webdriver
```

Comment:
How about splitting the conftest into two files. One for the server setup and one for the selenium?

Reply:
I think it's good to have it in one file, because ideally the connection between the server and the `wdriver` fixture should be an "implementation detail" (whatever that means for test infrastructure) -- essentially, the http server is just a means to an end of loading the demo into the webdriver.